### PR TITLE
[GSIP-143] Add ModuleStatus UI to Server Status page

### PR DESCRIPTION
**Pull request by mtCarto**

### CODE REVIEW

File: src/web/core/src/main/java/org/geoserver/web/admin/AbstractStatusPage.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<html xmlns:wicket="http://wicket.apache.org/">
+<body>
+<wicket:extend>
+    <p>
+        <div wicket:id="tabs" class="tabpanel"></div>
+        <wicket:child/>
+    </p>
+</wicket:extend>
+</body>
+</html>
```
Comment (bradh) [inline src/web/core/src/main/java/org/geoserver/web/admin/AbstractStatusPage.html]: Missing newline at end of file.

File: src/web/core/src/main/java/org/geoserver/web/admin/ModuleStatusPanel$MessagePanel.html [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  <body>
+    <wicket:panel>
+      <b><wicket:message key="moduleName"></wicket:message>: </b><span wicket:id="name"></span><br>
+      <b><wicket:message key="moduleStatus"></wicket:message>: </b><span wicket:id="module"></span><br>
+      <b><wicket:message key="moduleVersion"></wicket:message>: </b><span wicket:id="version"></span><br>
+      <b><wicket:message key="moduleComponent"></wicket:message>: </b><span wicket:id="component"></span><br>
+      <b><wicket:message key="moduleMessage"></wicket:message>: </b><pre wicket:id="msg"></pre>
+    </wicket:panel>
+  </body>
+</html>
```
Comment (bradh) [inline src/web/core/src/main/java/org/geoserver/web/admin/ModuleStatusPanel$MessagePanel.html]: Missing newline at end of file.

File: src/web/core/src/main/java/org/geoserver/web/admin/ModuleStatusPanel.html [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                    <td><span wicket:id="enabled"></span></td>
+                    <td><span wicket:id="component"></span></td>
+                    <td><span wicket:id="version"></span></td>
+               </tr>
+    </wicket:container>
+    </table>
+    <div wicket:id="popup"></div>
+</wicket:panel>
+</body>
+</html>
```
Comment (bradh) [inline src/web/core/src/main/java/org/geoserver/web/admin/ModuleStatusPanel.html]: Missing newline at end of file.

File: src/web/core/src/main/java/org/geoserver/web/admin/StatusPanel.java [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+        values.put(KEY_COVERAGEACCESS_MAX_POOL_SIZE, Integer.toString(coverageAccess.getMaxPoolSize()));
+        values.put(KEY_COVERAGEACCESS_KEEP_ALIVE_TIME, Integer.toString(coverageAccess.getKeepAliveTime()));
+
+        values.put(KEY_UPDATE_SEQUENCE, Long.toString(geoServerInfo.getUpdateSequence()));
+        values.put(KEY_JAVA_RENDERER, checkRenderer());
+    }
+
+   /**
+     * Retrieves the GeoServer data directory
+ * @param gsApp
```
Comment (bradh) [inline src/web/core/src/main/java/org/geoserver/web/admin/StatusPanel.java]: param doesn't appear.

**Comment by mtCarto** [comment]
@tbarsballe mind taking a quick final look over this?

File: doc/en/user/source/configuration/status.rst [hunk 5]
```diff
@@ -61,6 +61,41 @@ The following table describes the current status indicators.
      - GeoServer keeps in memory all of its configuration data. If for any reason that configuration information has become stale (e.g., an external utility has modified the configuration on disk) the "Reload" button will force GeoServer to reload all of its configuration from disk.
   
 
+Module Status
+-------------
```
Comment (tbarsballe) [inline doc/en/user/source/configuration/status.rst]: We have a header for module status here, but no header for the default tab

File: doc/en/user/source/configuration/status.rst [hunk 6]
```diff
@@ -61,6 +61,41 @@ The following table describes the current status indicators.
...
+
+The modules tab provides a summary of the status of all installed modules in the running server. 
+
+.. figure:: img/module_status.png
+   
+   Module Status
+   
+Module Status Field Descriptions
+--------------------------------
```
Comment (tbarsballe) [inline doc/en/user/source/configuration/status.rst]: This should probably be a nested header, not same level as Module Status, above

File: doc/en/user/source/configuration/status.rst [hunk 7]
```diff
@@ -61,6 +61,41 @@ The following table describes the current status indicators.
...
+     - (Optional) The version of the installed module
+   * - Message (popup)
+     - (Optional) status message such as what Java rendering engine is in use, or the library path if the module/driver is unavailable
+
+.. figure:: img/module_popup.png
+
+   Module Status popup
+
 Timestamps Field Descriptions
 -----------------------------
```
Comment (tbarsballe) [inline doc/en/user/source/configuration/status.rst]: Does this actually exist?

Comment (mtCarto) [inline doc/en/user/source/configuration/status.rst]: Waiting on response from mailing list about removing this section.

**Pull request merged**